### ISSUE TITLE
Difficulty in running inference for NCPDNet

### ISSUE THREAD

**Issue by jainspoornima** [zaccharieramzi/fastmri-reproducible-benchmark#175]
Hello, I trained the NCPDNet for 70 epochs using the following command on the FastMRI singlecoil knee dataset -
```
%cd /home/... /fastmri-reproducible-benchmark/fastmri_recon/training_scripts
!python -m nc_train
```

```
...
...
...
Epoch 68/70
973/973 [==============================] - 446s 459ms/step - loss: 7.6199 - mean_squared_error: 117.9911 - keras_psnr: 27.7218 - keras_ssim: 0.5965 - val_loss: 8.5947 - val_mean_squared_error: 151.7082 - val_keras_psnr: 28.6595 - val_keras_ssim: 0.6432
Epoch 69/70
973/973 [==============================] - 447s 459ms/step - loss: 7.5934 - mean_squared_error: 119.1959 - keras_psnr: 27.7173 - keras_ssim: 0.5944 - val_loss: 7.8597 - val_mean_squared_error: 121.3948 - val_keras_psnr: 30.1936 - val_keras_ssim: 0.6536
Epoch 70/70
973/973 [==============================] - 447s 459ms/step - loss: 7.6708 - mean_squared_error: 122.2345 - keras_psnr: 27.7937 - keras_ssim: 0.6002 - val_loss: 7.8543 - val_mean_squared_error: 116.3104 - val_keras_psnr: 27.3648 - val_keras_ssim: 0.6160
```

In the training, the loss is decreasing over the epochs, indicating learning.

And then ran the inference over it using the following command -
```
%cd /home/... /fastmri-reproducible-benchmark/fastmri_recon/evaluate/scripts
!python -m nc_eval
```
But the inference takes 6-7 hours (comparable to time taken for training for 70 epochs) and gives Weird PSNR and SSIM values -
```
...
...
...
2022-04-16 22:39:51.769981: W tensorflow/stream_executor/gpu/redzone_allocator.cc:314] Internal: Failed to launch ptxas
Relying on driver to perform ptx compilation. 
Modify $PATH to customize ptxas location.
This message will be only logged once.
  0%|                                                   | 0/199 [00:00<?, ?it/s]/home/poornima/Downloads/fastmri-reproducible-benchmark/fastmri_recon/evaluate/metrics/np_metrics.py:30: FutureWarning: `multichannel` is a deprecated argument name for `structural_similarity`. It will be removed in version 1.0. Please use `channel_axis` instead.
  return compare_ssim(gt.transpose(1, 2, 0), pred.transpose(1, 2, 0), multichannel=True, data_range=gt.max() - gt.min())

100%|██████████████████████████████████████| 199/199 [6:56:24<00:00, 125.55s/it]
dict_keys(['PSNR', 'SSIM'])
[-105.43021065982262, 3.040859079492017e-07]
```
I wished to ask if I am doing the inference and training for NCPDNet correctly and else, why does is the inference taking so long and giving output as weird metric values?

**Comment by zaccharieramzi** [comment]
Hi @jainspoornima ,

Sorry for the very belated answer.
You need to specify the run id of your NCPDNet when running the evaluation. It was printed by the training script, and is also available in the name of your checkpoint.
For example : 
```
python fastmri_recon/evaluate/scripts/nc_eval.py -r your-run-id -e 70
```

You have multiple options when training/evaluating non cartesian networks that you can see using the help function:
```
python fastmri_recon/evaluate/scripts/nc_train.py --help
```

**Comment by zaccharieramzi** [comment]
Closing this for clarity, but feel free to reopen if needed.
